### ISSUE TITLE
[iOS] Error: Unknown interface.

### ISSUE THREAD

**Issue by snapxtp** [Rapsssito/react-native-tcp-socket#6]
Hi;

Create server 127.0.0.1 port:9001 and create client but gives the following error.

var client = TcpSocket.createConnection({
            host:"127.0.0.1",
            port:9001
        }, (address) => {
            this.updateChatter('opened client on ' + JSON.stringify(address));
            client.write('Hello, server! Love, Client.');
        });

Error: Unknown interface. Specify valid interface by name (e.g. "en1") or IP address.
    at normalizeError (TcpSocket.js:414)
    at TcpSocket._onError (TcpSocket.js:307)
    at TcpSocket.js:245
    at RCTDeviceEventEmitter.emit (EventEmitter.js:189)
    at MessageQueue.__callFunction (MessageQueue.js:436)
    at MessageQueue.js:111
    at MessageQueue.__guard (MessageQueue.js:384)
    at MessageQueue.callFunctionReturnFlushedQueue (MessageQueue.js:110)
    at debuggerWorker.js:80

**Comment by Rapsssito** [comment]
Could you post your react native version, package version and platform?

**Comment by snapxtp** [comment]
server is working but the client mentioned above I get the error and it does not work.

"@babel/plugin-proposal-decorators": "^7.6.0",
 "@react-native-community/async-storage": "^1.6.3",
 "babel-plugin-transform-decorators-legacy": "^1.3.5",
  "react": "16.9.0",
  "react-native": "0.61.2"

Platform : ios 13

**Comment by Rapsssito** [comment]
I'll have a look. Working on a solution!

**Comment by Rapsssito** [comment]
It looks like if you want to use the `127.0.0.1` address, you must set the client's `localAddress` to `127.0.0.1`, too.

**Comment by snapxtp** [comment]
Sorry I didn't work on this. I get the same error when I create a new server using device ip. Server is running but I can't create a client.
Thank you

**Comment by Rapsssito** [comment]
I mean, you have to **create the `client`** with a `localAddress = 127.0.0.1`  if you want to use that address.

**Comment by snapxtp** [comment]
My Code;

export default class SocketServer {
    serverPort = 9001;
    serverHost = "127.0.0.1";
    server = null;
    client = null;
    startServer() {
        this.server = TcpSocket.createServer(async (socket) => {
            console.log(JSON.stringify(socket.address()))
            socket.on('data', async (data) => { 
                socket.write([
                    'HTTP/1.1 200 OK',
                    'Content-Type: application/json; charset=UTF-8',
                    'Content-Encoding: UTF-8',
                    'Accept-Ranges: bytes',
                    'Connection: keep-alive'
                ].join('\n') + '\n\n');
                socket.end();

            });
            socket.on('error', (error) => {
                console.log('Socket Error ',error)
            });
            socket.on('close', (error) => {
                console.log('Socket closing...',error)
                 console.log(error)
            });
        }).listen(this.serverPort, this.serverHost, () => {
            //this.updateChatter('opened server on ' + JSON.stringify(server.address()));
        });

        this.server.on('error', (error) => {
            console.log('Server error ',error)
        });

        this.server.on('close', () => {
            alert('Server closed')
        });
    }

    startClient(){
        this.client = TcpSocket.createConnection({
            host:this.serverHost,
            port:this.serverPort
        });

        client.on('data', function(data) {
        console.log('message was received', data);
        });

        client.on('error', function(error) {
        console.log(error);
        });

        client.on('close', function(){
        console.log('Connection closed!');
        });

        // Write on the socket
        client.write("Hello server!");

        // Close socket
        client.destroy();
    }
}

**Comment by Masoodt** [comment]
@snapxtp @Rapsssito install from my fork, i've a temporary fix
yarn add https://github.com/Masoodt/react-native-tcp-socket/tarball/master

**Comment by Rapsssito** [comment]
The issue is related to the library used in the iOS part (CocoaAsyncSocket), it cannot connect to `"127.0.0.1"` without specifying a localAddress to bind to.
```javascript
var client = TcpSocket.createConnection({
host:"127.0.0.1",
port:9001,
localHost:"127.0.0.1", // Add this
}, (address) => {
this.updateChatter('opened client on ' + JSON.stringify(address));
client.write('Hello, server! Love, Client.');
});
```

**Comment by Pinwheeler** [comment]
Turns out I'm getting this with my server's IP address as well (not 127.0.0.1)

It seems to be only happening on iOS and possibly only iOS simulator. I'll see if setting a local address with get around this problem but just wanted to give a heads up.

**Comment by Rapsssito** [comment]
This issue may be fixed in **[3.2.7](https://github.com/Rapsssito/react-native-tcp-socket/releases/tag/v3.2.7)**.
